src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -147,7 +147,69 @@ public RestAzureNS.AzureOperationResponse TriggerBackup()
 
         public RestAzureNS.AzureOperationResponse TriggerRestore()
         {
-            throw new NotImplementedException();
+            string vaultName = (string)ProviderData[VaultParams.VaultName];
+            string resourceGroupName = (string)ProviderData[VaultParams.ResourceGroupName];
+            CmdletModel.AzureFileShareRecoveryPoint recoveryPoint = ProviderData[RestoreBackupItemParams.RecoveryPoint]
+                as CmdletModel.AzureFileShareRecoveryPoint;
+            string storageAccountId = ProviderData[RestoreBackupItemParams.StorageAccountId].ToString();
+            string storageAccountLocation = ProviderData.ContainsKey(RestoreBackupItemParams.StorageAccountLocation) ?
+                ProviderData[RestoreBackupItemParams.StorageAccountLocation].ToString() : null;
+            string copyOptions = (string)ProviderData[RestoreFSBackupItemParams.ResolveConflict];
+            string sourceFilePath = ProviderData.ContainsKey(RestoreFSBackupItemParams.SourceFilePath) ?
+                (string)ProviderData[RestoreFSBackupItemParams.SourceFilePath] : null;
+            string sourceFileType = ProviderData.ContainsKey(RestoreFSBackupItemParams.SourceFileType) ?
+                (string)ProviderData[RestoreFSBackupItemParams.SourceFileType] : null;
+            string targetStorageAccountName =
+                ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetStorageAccountName) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetStorageAccountName] : null;
+            string targetFileShareName = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetFileShareName) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetFileShareName] : null;
+            string targetFolder = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetFolder) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetFolder] : null;
+            string targetStorageAccountId = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetStorageAccountId) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetStorageAccountId] : null;
+            string targetStorageAccountLocation = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetStorageAccountLocation) ?
+                ProviderData[RestoreFSBackupItemParams.TargetStorageAccountLocation].ToString() : null;
+
+            List<RestoreFileSpecs> restoreFileSpecs = null;
+            if (sourceFilePath != null || targetFolder != null)
+            {
+                RestoreFileSpecs restoreFileSpec = new RestoreFileSpecs();
+                restoreFileSpec.Path = sourceFilePath;
+                restoreFileSpec.TargetFolderPath = targetFolder;
+                restoreFileSpec.FileSpecType = sourceFileType;
+                restoreFileSpecs = new List<RestoreFileSpecs>();
+                restoreFileSpecs.Add(restoreFileSpec);
+            }
+
+            TargetAFSRestoreInfo targetDetails = null;
```

Comment:
Same comment as above. Should we throw error when the right parameters are not passed? What is PM's direction on this?

Reply:
If it's an original location restore, targetDetails are to be kept null and that's how the service API wants too

Reply:
Same comment above. Resolving this one.